backend/src/server.ts

```diff
@@ -50,12 +50,14 @@ class Server {
   }
 
   async start() {
+    const preferences = Preferences.get()
+    const HOST = preferences.disabledLocalNetwork ? 'localhost' : '0.0.0.0'
```

Comment:
You can get these constants from the shared/constants.ts file
```suggestion
    const HOST = preferences.disabledLocalNetwork ? IP_PRIVATE : IP_OPEN
```